src/datasets/metric.py

```diff
@@ -435,13 +436,20 @@ def add_batch(self, *, predictions=None, references=None):
             self._init_writer()
         try:
             self.writer.write_batch(batch)
-        except pa.ArrowInvalid:
-            raise ValueError(
-                f"Predictions and/or references don't match the expected format.\n"
-                f"Expected format: {self.features},\n"
-                f"Input predictions: {predictions},\n"
-                f"Input references: {references}"
-            ) from None
+        except pa.ArrowInvalid as e:
+            match = re.match(r"Column 1 named references expected length (\d+) but got length (\d+)", str(e))
+            if match is not None:
+                error_msg = (
+                    f"Mismatch in the number of predictions ({match.group(1)}) and references ({match.group(2)})"
+                )
+            else:
+                error_msg = (
+                    f"Predictions and/or references don't match the expected format.\n"
+                    f"Expected format: {self.features},\n"
+                    f"Input predictions: {predictions},\n"
+                    f"Input references: {references}"
```

Comment:
Maybe let's not print the full list of predictions and references ? It could be very very long ^^'

Reply:
Yes, I agree. We can implement a simple summarization similar to NumPy arrays. 

Currently, [NumPy](https://numpy.org/doc/stable/reference/generated/numpy.set_printoptions.html#numpy.set_printoptions) triggers summarization on arrays with more than 1000 elements. Do you have a threshold in mind for our use case?

Reply:
We can just crop at 5 elements IMO

Reply:
Let's make it 6 to have an equal number of elements on both sides of the summarized list if `len(batch) == 6`. 😃